lib/puppet/util/windows/user.rb

```diff
@@ -52,19 +60,18 @@ def check_token_membership
   module_function :check_token_membership
 
   def password_is?(name, password)
-    # Windows API LogonUserW always disallows this with error 1327:
-    # Logon failure: user account restriction. Possible reasons are blank passwords
-    # not allowed, logon hour restrictions, or a policy restriction has been enforced.
-    if password.nil? || password == ''
-      raise Puppet::Util::Windows::Error.new(
```

Comment:
After looking at this a bit more closely, a `nil` value may be passed successfully for a user with a blank password... however, for it to work properly, it also requires changing how the pointer is passed to FFI:

```ruby
password.nil? ? FFI::Pointer::NULL : wide_string(password)
```